dual/provider/provider.go

```diff
@@ -0,0 +1,139 @@
+package provider
+
+import (
+	"context"
+	"errors"
+
+	"github.com/ipfs/go-cid"
+	logging "github.com/ipfs/go-log/v2"
+	dht "github.com/libp2p/go-libp2p-kad-dht"
+	"github.com/libp2p/go-libp2p-kad-dht/dual"
+	"github.com/libp2p/go-libp2p-kad-dht/provider"
+	"github.com/libp2p/go-libp2p-kad-dht/provider/datastore"
+	mh "github.com/multiformats/go-multihash"
+)
+
+var _ provider.DHTProvider = &SweepingProvider{}
+
+var rLogger = logging.Logger("dht/dual/provider")
```

Comment:
(note about trying to reduce the number of logging facilities in general) I wonder if we can use `dht/dual` in general for this dht implementation. It appears that the parent module doesn't log at all though. There might be also good reasons not to, so it's just a comment.